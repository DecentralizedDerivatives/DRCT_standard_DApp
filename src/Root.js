import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from './reducers';

const middlewares = [thunk];
let composeEnhancer = compose;

if (process.env.NODE_ENV === 'development') {
  middlewares.push(createLogger());
  composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
}

const Root = ({ children, initialState = {} }) => {
  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancer(applyMiddleware(...middlewares))
  );

  return <Provider store={store}>{children}</Provider>;
};

export default Root;
