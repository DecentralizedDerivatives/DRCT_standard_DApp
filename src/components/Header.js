import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
// Dropdown disabled pending separate routes for dropdown items

// Use named export for unconnected component for testing
export class Header extends Component {
  constructor() {
    super();
    this.state = {
      showMobileNav: false
    }
  }
  handleNavClick = (e) => {
    this.setState({showMobileNav: false});
  }
  handleTermsClick = (e) => {
    e.preventDefault();
    this.handleNavClick(e);
    if (this.props.showTerms) { this.props.showTerms(); }
  }
  hamburgerClick = (e) =>{
    this.setState({showMobileNav: !this.state.showMobileNav});
  }
  render() {
    return (
      <nav id="header-nav">
        <div onClick={this.hamburgerClick} className={'hamburger-btn ' + (this.state.showMobileNav ? 'hamburger-btn-x' : '')}>
          <div></div>
          <div></div>
          <div></div>
        </div>
        <Link onClick={this.handleNavClick} to="/">
          <img
            className="nav-logo"
            src="./dda-logo.png"
            alt="logo"
            height="30px"
            width="30px"
          />
        </Link>
        <ul id='mobile-nav' className={this.state.showMobileNav ? 'show-nav' : 'hide-nav'}>
          <li>
            <Link onClick={this.handleNavClick} className={"nav-link"} to="/portfolio">Portfolio</Link>
          </li>
          <li>
            <Link onClick={this.handleNavClick} className={"nav-link"} to="/bulletin">Bulletin</Link>
          </li>
          <li>
            <Link onClick={this.handleNavClick} className={"nav-link"} to="/how-to">How To</Link>
          </li>
          <li>
            <p onClick={this.handleTermsClick} className="nav-link">Terms</p>
          </li>
        </ul>
        <div className="connection-status">
          {this.props.isConnected && this.props.whiteListed && (
            <div className="connected">
              Connected
                    <span style={{ color: 'green', paddingLeft: '5px' }}>
                <i className="fas fa-circle" />
              </span>
            </div>
          )}

          {this.props.isConnected && !this.props.whiteListed && (
            <div className="connected">
              Connected
                    <span style={{ color: 'yellow', paddingLeft: '5px' }}>
                <i className="fas fa-circle" />
              </span>
            </div>
          )}

          {!this.props.isConnected && (
            <div className="connected">
              Not Connected
                    <span style={{ color: 'red', paddingLeft: '5px' }}>
                <i className="fas fa-circle" />
              </span>
            </div>
          )}
        </div>
      </nav>
    );
  }
}

Header.propTypes = {
  showTerms: PropTypes.func,
  isConnected: PropTypes.bool.isRequired,
  whiteListed: PropTypes.bool.isRequired
};

export default Header;
