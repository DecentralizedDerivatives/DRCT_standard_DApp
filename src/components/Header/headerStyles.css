.connected {
  color: #fff;
  font-size: 16px;
  font-weight: 400;
  letter-spacing: 1px;
}

.link-text {
  color: #fff;
  font-size: 16px;
  font-weight: 400;
  letter-spacing: 1px;
}

.link-text-active {
  color: #114770;
  font-size: 16px;
  font-weight: 400;
  letter-spacing: 1px;
}

.header-button {
  height: 80px;
  border-right: 1px solid #27597e;
  padding: 0px 30px;
  border-radius: 0px;
}

.header-button-active {
  height: 80px;
  border-right: 1px solid #27597e;
  padding: 0px 30px;
  border-radius: 0px;
  background-color: #fff;
}

.header-button-active:hover {
  background-color: #fff;
}

.header-link {
  text-decoration: none;
}

@media only screen and (max-width: 600) {
  .header-button {
    display: none;
  }

  .header-button-active {
    display: none;
  }

  .header-link {
    display: none;
  }
}

.lens-icon {
  position: relative;
  margin-left: 6px;
  top: 2px;
  height: 13px;
  width: 13px;
}

.lens-on {
  fill: #30ad63;
}

.lens-off {
  fill: #ff4136;
}

.menu-icon: {
  width: 30px;
  height: 30px;
}

.menu-button {
  display: flex;
  align-self: center;
  margin-left: 10px;
}
