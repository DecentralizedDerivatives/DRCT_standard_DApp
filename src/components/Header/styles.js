const button = {
  height: '90px',
  borderRight: '1px solid #27597E',
  padding: '0px 30px',
  borderRadius: '0px',
};

const text = theme => ({
  color: theme.colors.white,
  fontSize: theme.fonts.size.header,
  fontWeight: theme.fonts.weight.medium,
  letterSpacing: '1px',
});

const lens = {
  position: 'relative',
  marginLeft: '6px',
  top: '2px',
  height: '13px',
  width: '13px',
};

export default theme => ({
  appBar: {
    backgroundColor: theme.colors.blue,
    height: '90px',
    minWidth: '832px',
  },
  linkText: {
    ...text(theme),
  },
  linkTextActive: {
    ...text(theme),
    color: theme.colors.blue,
  },
  connected: {
    ...text(theme),
  },
  button,
  buttonActive: {
    ...button,
    backgroundColor: theme.colors.white,
    '&:hover': {
      backgroundColor: theme.colors.white,
    },
  },
  container: {
    height: '100%',
  },
  lens: {
    ...lens,
    fill: theme.colors.green,
  },
  lensOff: {
    ...lens,
    fill: theme.colors.red,
  },
  connectedContainer: {
    marginRight: '20px',
    height: '100%',
  },
  logo: {
    width: '60px',
  },
  link: {
    textDecoration: 'none',
  },
});
