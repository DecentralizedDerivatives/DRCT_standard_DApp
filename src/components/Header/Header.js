import React, { Component } from 'react';
import { withRouter } from 'react-router';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import AppBar from 'material-ui/AppBar';
import Grid from 'material-ui/Grid';
import withStyles from 'material-ui/styles/withStyles';
import styles from './styles';
import './headerStyles.css';
import CreateContract from '../CreateContract';
import CashOut from '../CashOut';
import HowTo from '../HowTo';
import NavDrawer from '../NavDrawer';

class Header extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    connected: PropTypes.bool.isRequired
  };
  constructor() {
    super();
    this.state = {
      previousActive: '',
      active: '',
      openCash: false,
      openCreate: false,
      openHow: false,
      connected: true,
      drawerOpen: false
    };
  }
  static getDerivedStateFromProps(nextProps, prevState) {
    const url = nextProps.location.pathname.replace('/', '');

    return {
      ...prevState,
      ...nextProps,
      active: url
    };
  }

  onClick = link => {
    link = link.toLowerCase();
    if (link === 'create_contract') {
      this.openCreateContract();
    } else if (link === 'cash_out') {
      this.openCashOut();
    } else if (link === 'how_to') {
      this.openHowTo();
    }
    this.setState({ active: link });
  };

  openCreateContract = () => {
    this.setState({ openCreate: true, previousActive: this.state.active });
  };

  closeCreateContract = () => {
    this.setState({
      openCreate: false,
      active: this.state.previousActive
    });
  };

  openCashOut = () => {
    this.setState({ openCash: true, previousActive: this.state.active });
  };

  closeCashOut = () => {
    this.setState({
      openCash: false,
      active: this.state.previousActive
    });
  };

  openHowTo = () => {
    this.setState({ openHow: true, previousActive: this.state.active });
  };

  closeHowTo = () => {
    this.setState({
      openHow: false,
      active: this.state.previousActive
    });
  };

  toggleMetamaskModal = () => this.setState({ metamask: !this.state.metamask });

  renderHeaderLinks = () => {
    const { classes } = this.props;
    const urls = [
      'portfolio',
      'portfolio',
      'bulletin',
      'create_contract',
      'cash_out',
      'how_to'
    ];
    return [
      'Logo',
      'My Portfolio',
      'Bulletin',
      'Create Contract',
      'Cash Out',
      'How To'
    ].map((link, i) => {
      const component = (
        <Grid key={link} item>
          <button
            className={
              this.state.active === urls[i] ? 'button-active' : 'button'
            }
            onClick={() => this.onClick(urls[i])}
          >
            <span
              key={link}
              className={
                this.state.active === urls[i] ? 'link-text-active' : 'link-text'
              }
            >
              {link}
            </span>
          </button>
        </Grid>
      );

      if (i === 0) {
        return (
          <Link className="header-link" to={`/${urls[i]}`} key={link}>
            <div className="logo">
              <img
                src="dda-logo.png"
                width="70"
                alt="Logo"
                height="70"
                className="header-link"
                style={{ marginTop: '7%', marginRight: '20%' }}
              />
            </div>
          </Link>
        );
      }

      if (i === 0) {
        return (
          <Link className="header-link" to={`/${urls[i]}`} key={link}>
            <div className="logo">
              <img
                src="dda-logo.png"
                width="70"
                height="70"
                alt="Home"
                className="header-link"
                style={{ marginTop: '7%', marginRight: '20%' }}
              />
            </div>
          </Link>
        );
      }

      if (i !== 3) {
        return (
          <Link className="header-link" to={`/${urls[i]}`} key={link}>
            {component}
          </Link>
        );
      }

      return component;
    });
  };

  handleDrawer = () => this.setState({ drawerOpen: !this.state.drawerOpen });

  render() {
    const { classes } = this.props;
    return (
      <AppBar className={classes.appBar} position="static">
        <Grid
          container
          className={classes.container}
          direction="row"
          alignItems="stretch"
          justify="space-between"
        >
          <Grid item>
            <Grid
              container
              className={classes.container}
              alignItems="stretch"
              direction="row"
            >
              <Grid className={classes.menuContainer} item>
                <button
                  className="menu-button"
                  aria-label="Menu"
                  onClick={this.handleDrawer}
                >
                  <i className="material-icons menu-icon">menu</i>
                </button>
              </Grid>
              {this.renderHeaderLinks()}
            </Grid>
          </Grid>

          <Grid item>
            <Grid
              container
              className={classes.connectedContainer}
              direction="row"
              alignItems="center"
            >
              <Grid item>
                <p className="connected">Connected</p>
              </Grid>

              <Grid item>
                <i
                  className={
                    this.state.connected
                      ? 'material-icons lens-icon lens-on'
                      : 'material-icons lens-icon lens-off'
                  }
                >
                  lens
                </i>
              </Grid>
            </Grid>
          </Grid>
        </Grid>

        <NavDrawer
          open={this.state.drawerOpen}
          handleDrawer={this.handleDrawer}
          onClick={this.onClick}
        />

        <CreateContract
          open={this.state.openCreate}
          toggle={this.closeCreateContract}
        />
        <HowTo open={this.state.openHow} toggle={this.closeHowTo} />
        <CashOut open={this.state.openCash} toggle={this.closeCashOut} />
      </AppBar>
    );
  }
}

export default withRouter(withStyles(styles)(Header));
