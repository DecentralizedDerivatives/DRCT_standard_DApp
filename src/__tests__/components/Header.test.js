import React from 'react';
import { shallow } from 'enzyme';
import Root from '../../Root';
import Header from '../../components/Header';

describe('<Header />', () => {
  describe('render()', () => {
    xit('renders the component', () => {
      const wrapper = shallow(
        <Root>
          <Header />
        </Root>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
