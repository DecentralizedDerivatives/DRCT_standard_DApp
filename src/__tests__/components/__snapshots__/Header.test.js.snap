// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render connected and whitelisted renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Root>
    <Header
      isConnected={true}
      whiteListed={true}
    />
  </Root>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Header
        isConnected={true}
        whiteListed={true}
      />,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "isConnected": true,
        "whiteListed": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Header
          isConnected={true}
          whiteListed={true}
        />,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isConnected": true,
          "whiteListed": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Header /> render connected not whitelisted renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Root>
    <Header
      isConnected={true}
      whiteListed={false}
    />
  </Root>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Header
        isConnected={true}
        whiteListed={false}
      />,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "isConnected": true,
        "whiteListed": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Header
          isConnected={true}
          whiteListed={false}
        />,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isConnected": true,
          "whiteListed": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Header /> render not connected renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Root>
    <Header
      isConnected={false}
      whiteListed={false}
    />
  </Root>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Header
        isConnected={false}
        whiteListed={false}
      />,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "isConnected": false,
        "whiteListed": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Header
          isConnected={false}
          whiteListed={false}
        />,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isConnected": false,
          "whiteListed": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
